test/_chainrules.jl

```diff
@@ -0,0 +1,34 @@
+@testset "chainrules.jl" begin
+
+    function FiniteDifferences.to_vec(k::AxisKeys.KaNda)
+        v, b = to_vec(k.data)
+        back(x) = wrapdims(b(x); AxisKeys.named_axiskeys(k)...)
+        return v, back
+    end
+
+    function FiniteDifferences.to_vec(k::KeyedArray)
+        v, b = to_vec(k.data)
+        back(x) = wrapdims(b(x), axiskeys(k)...)
+        return v, back
```

Comment:
Do we actually need these `to_vec` definitions?
FiniteDifferences.jl uses ProjectTo under-the-hood

Reply:
yes, I think so. Without them all of the tests fail.

Reply:
Have you tested this after adding `ProjectTo`? It might just work now.

Reply:
yes, I removed the `to_vec` functions, tested and they still fail. But not for the same reason.

>FiniteDifferences.jl uses ProjectTo under-the-hood

Where does it use it? I went looking but didn't find it.

Reply:
my mistake, it is used in `ChainRulesTestUtils.rand_tangent`

Reply:
And yeah, looking at FiniteDifferences i see that indeed right now it doesn't handle `AbstractArray`s in general.
Except by handling them as structs, which would make `Tangent`s.